skopt/dummy_opt.py

```diff
@@ -52,14 +70,32 @@ def dummy_minimize(func, dimensions, n_calls=100, random_state=None):
     """
     rng = check_random_state(random_state)
     space = Space(dimensions)
-    X = space.rvs(n_samples=n_calls, random_state=rng)
-
-    init_y = func(X[0])
+    if x0 is None:
+        x0 = []
+    x0 = list(x0)
+    if x0 and not isinstance(x0[0], list):
+        x0 = [x0]
```

Comment:
I find the code a bit easier to read, if it is restructured like this:

```python
if x0 is None:
    x0 = []
elif not isinstance(x0[0], list):
    x0 = [x0]

if not isinstance(x0, list):
    raise ValueError("Expected x0 to be a list, but got %s" % type(x0))

```

WDYT: Note we can decide about support for `x0` to be a numpy array in a follow-up PR